README.md

```diff
@@ -1,2 +1,128 @@
-# raincatcher
-Feedhenry Raincatcher Core Repository
+# RainCatcher
+Feedhenry RainCatcher Core Repository
+
+## About RainCatcher
+
+## Developing
+
+RainCatcher supports and is tested on Node.JS@4 and should run on latest LTS.
+
+Check your Node.JS version running `node -v`
+
+### Setup
+
+```bash
+git clone https://github.com/feedhenry-raincatcher/raincatcher-core.git
+cd raincatcher
+npm install -g lerna
+npm install
+lerna bootstrap
+```
+
+## Repository folder structure
+
+This repository contains many subpackages managed through [Lerna](https://lernajs.io/), and they're
+contained in the following directories:
+
+<dl>
+  <dt>client/</dt>
+  <dd>Front-end packages, from API clients to user interface modules</dd>
+
+  <dt>cloud/</dt>
+  <dd>Cloud app packages, intended to be run on an Node.JS environment</dd>
+
+  <dt>demo/</dt>
+  <dd>Full-fledged Demo applications, showcasing the usage of multiple modules</dd>
+
+  <dt>examples/</dt>
+  <dd>Smaller examples and templates for other packages in the repository</dd>
+</dl>
+
+## Creating a new package
+
+In order to create a new package, we recommend duplicating
+[`examples/base`](./examples/base/README.md), which is a sample base that contains the skeleton
+expected of a new package. Refer to the linked README for more details.
+
+## Package structure
+
+### Directories
+
+<dl>
+  <dt>src/</dt>
+  <dd>Contains the TypeScript sources.</dd>
+
+  <dt>out/</dt>
```

Comment:
I just had to change that from lib to out in order to include the compiled unit tests.
Otherwise we'd either have to compile the unit tests for publishing inside the `test/` directory itself...

I'll actually look if that's a better option.